src/tools/perf/lib/libperf.c

```diff
@@ -1220,7 +1220,8 @@ static ucs_status_t ucp_perf_test_setup_endpoints(ucx_perf_context_t *perf,
     unsigned i;
 
     if ((group_size != 2) &&
-            !(perf->params.flags & UCX_PERF_TEST_FLAG_LOOPBACK)) {
+            !((group_size == 1) &&
+                        (perf->params.flags & UCX_PERF_TEST_FLAG_LOOPBACK))) {
```

Comment:
now, the error message does not fit the check
need to split to 2 checks and 2 err messages - one for p2p, another for loopback